File: ios/Sources/GetLocalIPPlugin/GetLocalIPPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(GetLocalIPPlugin)
public class GetLocalIPPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "GetLocalIPPlugin"
    public let jsName = "GetLocalIP"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "getLocalIP", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = GetLocalIP()

    @objc func getLocalIP(_ call: CAPPluginCall) {
        if let ip = implementation.getLocalIP() {
            call.resolve([
                "ip": ip
            ])
        } else {
            call.reject("Unable to determine local IP address")
        }
    }
}

File: android/src/main/java/eu/touchnstars/plugins/getlocalip/GetLocalIPPlugin.java
package eu.touchnstars.plugins.getlocalip;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "GetLocalIP")
public class GetLocalIPPlugin extends Plugin {

    private GetLocalIP implementation = new GetLocalIP();

    @PluginMethod
    public void getLocalIP(PluginCall call) {
        try {
            String ip = implementation.getLocalIP(getContext());
            
            JSObject ret = new JSObject();
            if (ip != null) {
                ret.put("ip", ip);
                call.resolve(ret);
            } else {
                call.reject("Unable to determine local IP address. Make sure you are connected to WiFi.");
            }
        } catch (Exception e) {
            call.reject("Error getting local IP: " + e.getMessage(), e);
        }
    }
}
